File: src/typescript/key-generation/generate-key/generate-private-key.ts
import { createWaxFoundation } from '@hiveio/wax';

const waxApi = await createWaxFoundation();
const accountName = "your-account";
const role = "active"; // roles can be 'active', 'owner', 'posting', or 'memo'
// Important notice!!!
// The master password should always be a truly random and secure value, such
// as one retrieved from the JS crypto interface.
// We used Math.random() here just for convenience.
const masterPassword = Math.random().toString();

// Generating a new private key from a password
const privateKeyData = waxApi.getPrivateKeyFromPassword(accountName, role, masterPassword);

console.log(`Associated Public Key: ${privateKeyData.associatedPublicKey}`);
console.log(`WIF Private Key: ${privateKeyData.wifPrivateKey}`);

File: src/rust/key_generation/generate_key/generate_private_key.rs
use wax::prelude::*;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let wax_api = create_wax_foundation(None);
    let account_name = "your-account";
    let role = "active"; // roles can be 'active', 'owner', 'posting', or 'memo'
    // Important notice!!!
    // The master password should always be a truly random and secure value.
    let master_password = wax_api.generate_private_key()?;

    // Generating a new private key from a password
    let private_key_data = wax_api.get_private_key_from_password(
        account_name,
        role,
        &master_password,
    )?;

    println!(
        "Associated Public Key: {}",
        private_key_data.associated_public_key
    );
    println!("WIF Private Key: {}", private_key_data.wif_private_key);

    Ok(())
}
